src/test/java/jep/test/TestPyBuiltins.java

```diff
@@ -0,0 +1,336 @@
+package jep.test;
+
+import java.util.Set;
+
+import jep.Interpreter;
+import jep.SharedInterpreter;
+import jep.JepException;
+import jep.python.PyBuiltins;
+import jep.python.PyCallable;
+import jep.python.PyObject;
+
+public class TestPyBuiltins {
+
+    private Interpreter interp;
+
+    private PyBuiltins builtins;
+
+    /* An instance of the object type */
+    private PyObject object;
+    
+    /* The object type */
+    private PyCallable objectType;
+
+    /* A type that extends object */
+    private PyCallable subtypeOne;
+
+    /* Another type that extends object */
+    private PyCallable subtypeTwo;
+
+    /* An instance of one of the subtypes above*/
+    private PyObject subobject;
+
+    /* A dict for testing exec and eval */
+    private PyObject globals;
+
+    /* A dict for testing exec and eval */
+    private PyObject locals;
+
+    /* Set to a non-null value to fail the test */
+    private String failure;
+
+    private boolean testCallable() {
+        if (!builtins.callable(objectType)){
+            failure = "callable builtin returns object is not callable";
+            return false;
+        }
+        if (builtins.callable(object)){
+            failure = "callable builtin returns an object is callable";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testCompile() {
+        PyObject code = builtins.compile("1 + 1", "<string>", "eval");
+        Number result = (Number) builtins.eval(code, globals);
+        if (result.intValue() != 2){
+            failure = "compile builtin does not return 1 + 1 = 2";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testDelAttr() {
+        subobject.setAttr("name", "value");
+        builtins.delattr(subobject, "name");
+        Set<String> dir = Set.of(builtins.dir(subobject));
```

Comment:
It appears that `of` was introduced after Java 8, hence failing Appveyor.  I would prefer if we still supported Java 8.